ministryofjustice/correspondence_tool_staff
# CT-3053 Added original linkage between Offender SAR and Complaint Case

## Description

This PR is for 
- Mainly adding original case link between offender sar and complaint case 
- Complete  the first 2 step of offender complaint creation work (link to a offender sar case and confirm the case chosen), copy over the subject details and third party detail 
- Restructured controller and OffenderComplaintForm to make the re-use easier (IMO :))
- Removed some redundant codes from some of files
- Minor bug fix for case type filter

## Self-review checklist


### Screenshots


### Related JIRA tickets

https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=25&modal=detail&selectedIssue=CT-3259
https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=25&modal=detail&selectedIssue=CT-3200

### Deployment


### Manual testing instructions

**NOTES: Make sure you give Branston team the permission to access Offender SAR complaint type as responder**
1 - Login as Branston user
2 - Click "Create case"
3 - Choose complaint case type
4 - type the case number 
5 - follow the steps
6 - check the details
